test/specs/actions/integrations.spec.js

```diff
@@ -24,62 +30,171 @@ describe('Integrations Actions', () => {
+
         updateUserSpy = sandbox.spy(updateUser);
         IntegrationsRewire('updateUser', updateUserSpy);
+
+        updateSMSAttributesSpy = sandbox.spy(updateSMSAttributes);
+        IntegrationsRewire('updateSMSAttributes', updateSMSAttributesSpy);
+
+        setTransferRequestCodeSpy = sandbox.spy(setTransferRequestCode);
+        IntegrationsRewire('setTransferRequestCode', setTransferRequestCodeSpy);
+
+        setWeChatQRCodeSpy = sandbox.spy(setWeChatQRCode);
+        IntegrationsRewire('setWeChatQRCode', setWeChatQRCodeSpy);
+
+        setViberQRCodeSpy = sandbox.spy(setViberQRCode);
+        IntegrationsRewire('setViberQRCode', setViberQRCodeSpy);
+
         handleConversationUpdatedStub = sandbox.stub().returnsAsyncThunk();
         IntegrationsRewire('handleConversationUpdated', handleConversationUpdatedStub);
 
+        subscribeFayeStub = sandbox.stub().returnsAsyncThunk();
+        IntegrationsRewire('subscribeFaye', subscribeFayeStub);
+
+        appUserId = hat();
         mockedStore = createMockedStore(sandbox, generateBaseStoreProps({
             user: {
-                _id: '1'
+                _id: appUserId
             }
         }));
-
     });
 
     afterEach(() => {
         sandbox.restore();
     });
 
+    function mockStartConversation(appUserId, conversationId) {
+        startConversationStub.callsFake(() => {
+            mockedStore.getState().user._id = appUserId;
+            mockedStore.getState().conversation._id = conversationId;
+
+            return () => Promise.resolve();
+        });
+    }
 
     describe('linkSMSChannel', () => {
         let conversationId;
+        let createdClient;
 
         beforeEach(() => {
-            httpStub.returnsAsyncThunk({
-                value: {
-                    client: {}
-                }
-            });
-
             conversationId = hat();
-            mockedStore = createMockedStore(sandbox, generateBaseStoreProps({
-                conversation: {
-                    _id: conversationId
-                }
-            }));
+            createdClient = {
+                id: hat()
+            };
         });
 
-        it('should set the twilio integration to pending state', () => {
-            const {config: {appId}, user: {_id}} = mockedStore.getState();
-            return mockedStore.dispatch(integrationsActions.linkSMSChannel({
-                type: 'twilio',
-                phoneNumber: '+0123456789'
-            })).then(() => {
-                httpStub.should.have.been.calledWith('POST', `/apps/${appId}/appusers/${_id}/clients`, {
-                    criteria: {
-                        type: 'twilio',
-                        phoneNumber: '+0123456789'
-                    },
-                    confirmation: {
-                        type: 'prompt'
+        describe('when the user exists', () => {
+            beforeEach(() => {
+                httpStub.returnsAsyncThunk({
```

Comment:
Marc was doing that a while ago, and the first time I read it I was like "why wouldn't you just use `stub.resolves()`?", then I noticed that it's actually returning a _function_ which returns a promise when invoked - it wasn't immediately obvious what the function was doing there. I think `returnsAsyncThunk` hides some of that confusion, even if it brings confusion of it's own.

Both styles are kind of ugly, but I'd rather keep `returnsAsyncThunk` and `returnsSyncThunk` since that's what's used in all the other tests